### PR TITLE
use checkbox component in job listing specification

### PR DESCRIPTION
**Pull request by alexbowen**
https://dfedigital.atlassian.net/browse/TEVA-1065

## Changes in this PR:

- use checkbox component for subjects selector on job specification page (required some tweeks to css/js of component)
- remove sort checkboxes by checked/not checked behaviour (as agreed with UX)
- any other functionality/appearance should not be regressed

### CODE REVIEW

File: app/views/hiring_staff/organisations/show.html.haml [hunk 1]
```diff
@@ -10,7 +10,11 @@
       - else
         = link_to t('sign_in.organisation.change'), auth_dfe_callback_path, class: 'govuk-link'
 
-  .govuk-grid-column-full{ class: 'govuk-!-margin-top-5' }
+.govuk-grid-row{ class: 'govuk-!-margin-top-5' }
```
Comment (cpjmcquillan) [inline app/views/hiring_staff/organisations/show.html.haml:13]: Is this supposed to be here still?

Comment (alexbowen) [reply app/views/hiring_staff/organisations/show.html.haml:13]: good spot, no its not

File: app/views/shared/_checkbox_group.html.haml [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
-        :first
+.tv-checkbox__group
+  - if search
+    .search-input
+      %input.govuk-input.search-input__input{ placeholder: "Search" }
+      %button{type: "submit"}
+        %i.fa.fa-search
+  .govuk-form-group
+    .govuk-checkboxes.govuk-checkboxes--small
+      = form_for '', url: 'url', html: { method: 'post' } do |f|
```
Comment (cpjmcquillan) [inline app/views/shared/_checkbox_group.html.haml:10]: And here we just have...

`= f.govuk_collection_check_boxes attribute_name,
     options,
     value_method,
     text_method,
     hint_method`

File: app/views/hiring_staff/vacancies/job_specification/show.html.haml [hunk 3]
```diff
@@ -42,16 +42,9 @@
...
 
-        = f.govuk_collection_check_boxes :subjects,
-          subject_options,
-          :first,
-          :first,
-          :last,
-          small: true,
-          classes: 'checkboxes-with-scroll subjects-options js-disabled-border'
+        %div{ class: 'govuk-!-margin-bottom-6' }
+          = render partial: 'shared/checkbox_group', locals: { :search => true, :options => subject_options, :form => 'job_specification_form[subjects]' }
```
Comment (cpjmcquillan) [inline app/views/hiring_staff/vacancies/job_specification/show.html.haml:47]: Could we do...

`= render 'shared/checkbox_group', search: true, f: f, attribute_name: :subjects, options: subject_options, value_method: :first, text_method: :first, hint_method: :last`

File: app/views/shared/_checkbox_group.html.haml [hunk 4]
```diff
@@ -1,13 +1,8 @@
...
-        options,
-        :last,
-        :first
+.tv-checkbox__group
+  - if search
+    .search-input
+      %input.govuk-input.search-input__input{ placeholder: "Search" }
+      %button{type: "submit"}
+        %i.fa.fa-search
+  = f.govuk_collection_check_boxes attribute_name, options, value_method, text_method, hint_method
```
Comment (cpjmcquillan) [inline app/views/shared/_checkbox_group.html.haml:8]: Can we pass the small boolean through too please?

**Pull request merged**